tools/integration_tests/util/creds_tests/admin_creds.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Run tests for --key-file flag and GOOGLE_APPLICATION_CREDENTIALS env variable
+
+package creds_tests
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/mounting"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/mounting/static_mounting"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+func RunTestsForKeyFileAndGoogleApplicationCredentialsEnvVarSet(testFlagSet [][]string, m *testing.M) (successCode int) {
+	// Saving testBucket value for setting back after testing.
+	testBucket := setup.TestBucket()
+
+	// Set the testBucket value to the bucket belonging to a different project for testing credentials.
+	setup.SetTestBucket("integration-test-gcsfuse")
```

Comment:
Can you try if this works to avoid hard coding the bucket here:
1. Before test, remove storage permission for default SA only for the passed testBucket using https://cloud.google.com/storage/docs/gsutil/commands/iam
2. After test, give the permissions back to testBucket.

Reply:
I checked we cant't do this as we can't change the permissions of project service account. you can check it on pantheon also the edit button is disable for the same.